src/tests/EShopworld.WorkerProcess.IntegrationTests/WorkerLeaseTests.cs

```diff
@@ -210,15 +209,303 @@ public async Task TestMultipleWorkLeasesSamePriority()
             }
         }
 
-        private void SetupWorkerLeases(int leaseCount, Func<int,int> assignPriorityFunc)
+        [Fact, IsIntegration]
+        public async Task TestWorkLease_WhenLowerPriorityStartsLeaseBeforeHigherPriority_TheHighestPriorityShouldAlwaysWin()
+        {
+            //arrange
+            List<IWorkerLease> expired = new List<IWorkerLease>();
+            ManualResetEvent manualResetEvent = new ManualResetEvent(false);
+            var leaseStore = _serviceProvider.GetService<ILeaseStore>();
+
+            Guid? winner = null;
+            void OnWorkerProcessOnLeaseAllocated(object sender, LeaseAllocatedEventArgs args)
+            {
+                var wp = sender as WorkerLease;
+                winner = wp.InstanceId;
```

Comment:
you will have multiple threads trying to update that value - don't you need some critical section/lock here?

Reply:
sure included in a lock